src/alignment_utils.py

```diff
@@ -0,0 +1,84 @@
+        filename_timestamps = map(
+            lambda x: int(x.split('.')[0]),
+            filter(
+                lambda x: x.split('.')[1] in ALLOWED_EXTENSIONS,
+                os.listdir(target_dir)
+            )
+        )
+        filename_timestamps.sort()
+        extension = os.listdir(target_dir)[0].split('.')[1]
+        timestamp = filename_timestamps[0]
+        # obtain delta with the first filename timestamp and reference timestamp
+        delta = ref_timestamp - timestamp
+
+        print("Aligning with sequence %d, timestamps %d - %d" % (seq, timestamp, ref_timestamp))
+        _align(target_dir, filename_timestamps, extension, delta)
+
+
+def align_by_delta(time_ref, target_dir, video_path):
+    # load frame timestamps csv, rename frames according to it
+    video_root, video_filename = os.path.split(video_path)
+    video_name = video_filename.split('.')[0]
+    video_date = re.sub(r"VID_((\d|_)*)", r"\1", video_name)
+
+    with open(os.path.join(video_root, video_date, video_name + "_timestamps.csv")) as frame_timestamps_file:
+        filename_timestamps = map(
+            lambda x: int(x), frame_timestamps_file.readlines()
+        )
+        extension = os.listdir(target_dir)[0].split('.')[1]
+        for i, timestamp in enumerate(filename_timestamps):
+            os.rename(
+                os.path.join(target_dir, "frame-%d.png" % (i + 1)),
+                os.path.join(target_dir, str(timestamp) + "." + extension)
+            )
+        with open(time_ref, 'r') as time_ref_file:
+            values = time_ref_file.readline().split(',')
+            seq = int(values[0])
+            ref_timestamp = int(values[1])
+            timestamp = int(values[2])
+            # obtain delta with the info from time reference file
+            delta = ref_timestamp - timestamp
+
+            # align with delta
+            print("Aligning with sequence %d, timestamps %d - %d" % (seq, timestamp, ref_timestamp))
+            _align(target_dir, filename_timestamps, extension, delta)
+
+
+def _align(target_dir, filename_timestamps, extension, delta):
+    with open(os.path.join(target_dir, '_transformation_metainf.csv'), 'w') as transformation_file:
+        transformation_writer = csv.DictWriter(
+            transformation_file, delimiter=',', fieldnames=['seq', 'old_stamp', 'new_stamp']
+        )
+        transformation_writer.writeheader()
+        for seq, old_stamp in enumerate(filename_timestamps):
+            new_stamp = int(old_stamp) + delta
+            new_name = str(new_stamp) + "." + extension
+            old_name = str(old_stamp) + "." + extension
+            print("Old name: %s new name: %s" % (old_name, new_name))
+
+            transformation_writer.writerow({'seq': seq, 'old_stamp': old_stamp, 'new_stamp': new_stamp})
+            os.rename(
+                os.path.join(target_dir, old_name),
+                os.path.join(target_dir, new_name)
+            )
```

Comment:
PEP 8 likes newlines at the end of the files :) 

Btw, I've not tried this yet -- https://github.com/apps/pep8-speaks, but looks like an easy-to-setup checker, maybe we can try to turn it on for our repos.

Reply:
There is a newline at the end - same problem with github not displaying it